### DISCUSSION TITLE
Randomly pair two sets of stimuli

### DISCUSSION THREAD

**Discussion by OllieSayeed** [jspsych/jsPsych#2618] [Q&A] [answered]
Hi! I'm running an artificial language experiment where each word of the artificial language should be randomly paired with an image, consistently for each participant but randomized across participants. So I have a set of 30 images and 30 audio files to pair up at the beginning of the experiment.

What's the easiest way of doing this? I was thinking of making an array of images, an array of sound files, shuffling one of them with `jsPsych.randomization.shuffle` and then finding a way for each trial to refer to the ith image and ith sound file – I'm guessing there's a better way.

**Comment by jodeleeuw** [comment, marked as answer]
Hi @OllieSayeed 

I think your approach sounds reasonable. You could create a joint array out of the two arrays in order to take advantage of [timeline variables](https://www.jspsych.org/7.2/overview/timeline/#timeline-variables). Something like this:

```js
let arr_images = jsPsych.randomization.shuffle([...]);
let arr_audio = jsPsych.randomization.shuffle([...]);

let trials = [];
for(let i = 0; i<arr_images.length; i++){
	trials.push({image: arr_images[i], audio: arr_audio[i]});
}
```

**Reply by OllieSayeed** [reply]
Thank you! I'll try this.
